service/api.go

```diff
@@ -560,9 +562,27 @@ func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	errs := validator.CheckYamlStrict(input.ProjectYaml)
+	errs := validator.ValidationErrors{}
+	if input.ProjectID != "" {
+		projectRef, err := model.FindBranchProjectRef(input.ProjectID)
```

Comment:
Should use FindMergedProjectRef; we're introducing a feature that will allow users to specify default values at the repo level that will apply to all branches for that repo. FindBranchProjectRef doesn't use these default values though.